File: snow_pkg/src/openalea/SnowDepthTrans.py
# coding: utf8
from copy import copy

import numpy
from math import *

def model_SnowDepthTrans(Sdepth = 0.0,
         Pns = 100.0):
    """
     - Name: SnowDepthTrans -Version: 1.0, -Time step: 1
     - Description:
                 * Title: snow cover depth conversion
                 * Author: STICS
                 * Reference: -
                 * Institution: INRA
                 * Abstract: -
     - inputs:
                 * name: Sdepth
                               ** min : 0.0
                               ** default : 0.0
                               ** max : 500.0
                               ** uri : 
                               ** variablecategory : state
                               ** datatype : DOUBLE
                               ** inputtype : variable
                               ** unit : m
                               ** description : snow cover depth Calculation
                 * name: Pns
                               ** parametercategory : constant
                               ** min : 
                               ** datatype : DOUBLE
                               ** max : 
                               ** uri : 
                               ** default : 100.0
                               ** inputtype : parameter
                               ** unit : cm/m
                               ** description : density of the new snow
     - outputs:
                 * name: Sdepth_cm
                               ** min : 0.0
                               ** variablecategory : state
                               ** max : 500.0
                               ** uri : 
                               ** datatype : DOUBLE
                               ** unit : cm
                               ** description : snow cover depth in cm
    """

    Sdepth_cm = Sdepth * Pns
    return Sdepth_cm

File: snow_pkg/src/f90/SnowDepthTrans.f90
MODULE Snowdepthtransmod
    IMPLICIT NONE
CONTAINS

    SUBROUTINE model_SnowDepthTrans(Sdepth, &
        Pns, &
        Sdepth_cm)
        IMPLICIT NONE
        REAL, INTENT(IN) :: Sdepth
        REAL, INTENT(IN) :: Pns
        REAL, INTENT(OUT) :: Sdepth_cm
        !- Description:
    !            * Title: snow cover depth conversion
    !            * Author: STICS
    !            * Reference: -
    !            * Institution: INRA
    !            * Abstract: -
        !- inputs:
    !            * name: Sdepth
    !                          ** description : snow cover depth Calculation
    !                          ** inputtype : variable
    !                          ** variablecategory : state
    !                          ** datatype : DOUBLE
    !                          ** default : 0.0
    !                          ** min : 0.0
    !                          ** max : 500.0
    !                          ** unit : m
    !                          ** uri : 
    !            * name: Pns
    !                          ** description : density of the new snow
    !                          ** inputtype : parameter
    !                          ** parametercategory : constant
    !                          ** datatype : DOUBLE
    !                          ** default : 100.0
    !                          ** min : 
    !                          ** max : 
    !                          ** unit : cm/m
    !                          ** uri : 
        !- outputs:
    !            * name: Sdepth_cm
    !                          ** description : snow cover depth in cm
    !                          ** variablecategory : state
    !                          ** datatype : DOUBLE
    !                          ** min : 0.0
    !                          ** max : 500.0
    !                          ** unit : cm
    !                          ** uri : 
        Sdepth_cm = Sdepth * Pns
    END SUBROUTINE model_SnowDepthTrans

END MODULE
